packages/metro/src/ModuleGraph/worker/collectDependencies.js

```diff
@@ -251,6 +289,98 @@ function collectDependencies<TSplitCondition = void>(
   };
 }
 
+/** Extract args passed to the `require.context` method. */
+function getRequireContextArgs(
+  path: NodePath<CallExpression>,
+): [string, RequireContextParams] {
+  const args = path.get('arguments');
+
+  let directory: string;
+  if (
+    !Array.isArray(args) ||
+    args.length < 1 ||
+    args[0].node.type !== 'StringLiteral'
+  ) {
+    throw new InvalidRequireCallError(path);
+  } else {
+    directory = args[0].node.value;
+  }
+
+  // Default to requiring through all directories.
+  let recursive: boolean = true;
+  if (args.length > 1) {
+    if (args[1].node.type !== 'BooleanLiteral') {
+      throw new InvalidRequireCallError(path);
+    }
+    recursive = args[1].node.value;
+  }
+  // Default to all files.
+  let filter = /^\.\/.*$/;
+  if (args.length > 2) {
+    const argNode = args[2].node;
+    if (argNode.type === 'RegExpLiteral') {
+      filter = new RegExp(argNode.pattern, argNode.flags);
+    } else {
+      // TODO: Handle `new RegExp(...)` -- `argNode.type === 'NewExpression'`
+      throw new InvalidRequireCallError(path);
+    }
+  }
+  // Default to `sync`.
+  let mode: ContextMode = 'sync';
+  if (args.length > 3) {
+    const argNode = args[3].node;
+    if (argNode.type === 'StringLiteral') {
+      mode = getContextMode(argNode.value);
+    } else {
+      // TODO: Handle `new RegExp(...)` -- `argNode.type === 'NewExpression'`
+      throw new InvalidRequireCallError(path);
+    }
+  }
+
```

Comment:
Let's also be strict about unknown arguments.

```suggestion

  if (args.length > 4) {
    throw new InvalidRequireCallError(path);
  }

```